src/index.ts

```diff
@@ -184,8 +184,8 @@ function commit(input: HTMLTextAreaElement | HTMLInputElement, list: HTMLElement
   return true
 }
 
-function fireCommitEvent(target: Element): void {
-  target.dispatchEvent(new CustomEvent('combobox-commit', {bubbles: true}))
+function fireCommitEvent(target: Element, originalEvent?: MouseEvent): void {
+  target.dispatchEvent(new CustomEvent('combobox-commit', {bubbles: true, detail: {originalEvent}}))
```

Comment:
Nitpick; What do you think about taking in `details` as a argument so that we adhere more to the API of `CustomEvent`?

```suggestion
function fireCommitEvent(target: Element, detail: Record<string, unknown> | undefined = undefined): void {
  target.dispatchEvent(new CustomEvent('combobox-commit', {bubbles: true, detail}))
```

(☝🏻 I wrote this suggestion in the GitHub comment box so it might be wrong)